app/models/concerns/issues/canonical_data.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+module Issues::CanonicalData
+  extend ActiveSupport::Concern
+
+  included do
+    before_save :set_canonical_data, if: :data_changed?
```

Comment:
```suggestion
    serialize :canonical_data, JSON

    before_save :set_canonical_data, if: :data_changed?
```
Me vas a matar, pero me acordé que se puede hacer esto y Rails lo maneja "mágico" al to_json y from_json.

Reply:
uhhh genial esta!!! tampoco la tenia jajajaja